<launch>
  <!-- This configuration simulates a vehicle model and sensors. 
  The configuration does the following:
    1. Loads the simulator with the vehicle model
    2. Loads sensor drivers 
        - GPS
        - IMU
        - DVL -->
        
	<arg name="model" default="$(find labust_sim)/launch/model.yaml" />
  <arg name="external_altitude" default="false" />
	
    
	<!-- Dynamic simulation node -->
	<group ns="simulation">
    <!-- LTP origin Location -->
  	<param name="LocalOriginLat" value="42"/>
    <param name="LocalOriginLon" value="15"/>
    <!-- Start the geodetic navigation node -->
    <node pkg="labust_navigation" type="llnode" name="llnode" output="screen">
      <param name="LocalFixSim" value="true" />
    </node>
    <!-- Load the model parameters -->  
    <rosparam command="load" file="$(arg model)" />
     
    <node pkg="labust_sim" type="uvsim" name="uvsim">
			<param name="Rate" value="10" />
			<param name="ModelWrap" value="1" />
			<param name="publish_world" value="false" />
			<param name="publish_sim_base" value="true" />

      <!-- Disable internal allocation -->
			<param name="maxThrust" value="100" />
			<param name="minThrust" value="-100" />

      <!-- Map the input forces to the achieved -->
			<remap from="tauIn" to="tauAch" />
			<remap from="meas_ideal" to="position" />
		</node>
		
		<!-- IMU sensor simulation node -->
		<node pkg="labust_sim" type="imu_sim" name="imu_sim">
      <rosparam param="offset">[-0.008,0.575,-0.15]</rosparam>
      <rosparam param="orot">[0.0,90.0,180.0]</rosparam>
		</node>
        
		<!-- GPS sensor simulation node -->
		<node pkg="labust_sim" type="gps_sim" name="gps_sim">
      <rosparam param="offset">[-0.2115,0.0,-0.38]</rosparam>
      <rosparam param="orot">[0.0,0.0,0.0]</rosparam>
      <!-- Publish with 2 Hz -->
			<param name="gps_pub" value="2" />
      <remap from="fix" to="gps" />
    </node>
        
		<!-- DVL sensor simulation node -->
		<node pkg="labust_sim" type="dvl_sim" name="dvl_sim">
      <rosparam param="offset">[0.0,0.0,0.245]</rosparam>
      <rosparam param="orot">[0.0,0.0,180.0]</rosparam>
			<remap from="altitude" to="altitude_int" if="$(arg external_altitude)" />
		</node>        		
	</group>
</launch>
